employee-rostering-frontend/src/ui/pages/rotation/EditShiftTemplateModal.tsx

```diff
@@ -0,0 +1,315 @@
+                  .add(this.state.editedValue.startTime.minutes, "minutes").format("HH:mm") 
+                : 
+                undefined}
+              onChange={(v) => {
+                const parts = v.split(":");
+                this.setState(old => ({
+                  editedValue: {
+                    ...old.editedValue,
+                    startTime: {
+                      hours: parseInt(parts[0]),
+                      minutes: parseInt(parts[1])
+                    }
+                  }
+                }));
+              }}
+            />
+          </InputGroup>
+          <InputGroup>
+            <Label>End Day Offset</Label>
+            <TextInput
+              aria-label="End Day Offset"
+              type="number"
+              defaultValue={(this.state.editedValue.endDayOffset !== undefined)? String(
+                this.state.editedValue.endDayOffset + 1) : undefined}
+              min={1}
+              max={this.props.rotationLength? this.props.rotationLength : undefined}
+              onChange={(v) => {
+                this.setState(old => ({
+                  editedValue: {
+                    ...old.editedValue,
+                    endDayOffset: v? parseInt(v) - 1 : undefined
+                  }
+                }));
+              }}
+            />
+            <Label>End Time</Label>
+            <TextInput
+              aria-label="End Time"
+              defaultValue={this.state.editedValue.endTime? 
+                moment("2018-01-01T00:00").add(this.state.editedValue.endTime.hours, "hours")
+                  .add(this.state.editedValue.endTime.minutes, "minutes").format("HH:mm")
+                : 
+                undefined}
+              type="time"
+              onChange={(v) => {
+                const parts = v.split(":");
+                this.setState(old => ({
+                  editedValue: {
+                    ...old.editedValue,
+                    endTime: {
+                      hours: parseInt(parts[0]),
+                      minutes: parseInt(parts[1])
+                    }
+                  }
+                }));
+              }}
+            />
+          </InputGroup>
+          <InputGroup>
+            <Label>Rotation Employee</Label>
+            <TypeaheadSelectInput
+              aria-label="Employee"
+              emptyText="Unassigned"
+              defaultValue={this.props.shiftTemplate? this.props.shiftTemplate.rotationEmployee : undefined}
+              options={[undefined, ...this.props.employeeList]}
+              optionToStringMap={employee => employee? employee.name : "Unassigned"}
+              onChange={employee => this.setState(prevState => ({
+                editedValue: { ...prevState.editedValue, rotationEmployee: employee? employee : null }
+              }))}
+              optional
+            />
+          </InputGroup>
+        </Form>
+      </Modal>
+    );
+  }
+}
+
+// eslint-disable-next-line no-undef
```

Comment:
This doesn't seem to be needed.

Reply:
Confirmed; not needed (although Eslint get easily confused with interfaces and other object to the point it usually does complain about default exporting an interface/constant without that line).